desci-server/src/controllers/data/delete.ts

```diff
@@ -50,84 +40,61 @@ export const deleteData = async (req: Request, res: Response<DeleteResponse | Er
-    const { updatedDagCidMap, updatedRootCid } = await removeFileFromDag(
-      dataBucket.payload.cid,
-      cleanContextPath,
-      pathToDelete,
-    );
-
-    /*
-     ** Prepare updated refs
+    /**
+     * Remove draft node tree entries, add them to the cid prune list
      */
-    const existingDataRefs = await prisma.dataReference.findMany({
+    // debugger;
+    const entriesToDelete = await prisma.draftNodeTree.findMany({
       where: {
         nodeId: node.id,
-        userId: owner.id,
-        type: { not: DataType.MANIFEST },
+        OR: [
+          {
+            path: {
+              startsWith: path + '/',
+            },
+          },
+          {
+            path: path,
+          },
+        ],
       },
     });
 
-    //map existing ref neutral paths to the ref
-    const existingRefMap = existingDataRefs.reduce((map, ref) => {
-      map[neutralizePath(ref.path)] = ref;
-      return map;
-    }, {});
-
-    const newRefs = await prepareDataRefs(node.uuid, latestManifest, updatedRootCid);
-
-    // Find the dags that need updating
-    const dataRefUpdates: Partial<DataReference>[] = newRefs.map((newDataRef) => {
-      const newRefNeutralPath = neutralizePath(newDataRef.path);
-      const match = existingRefMap[newRefNeutralPath];
-      match.rootCid = updatedRootCid;
-      if (match) {
-        return { ...match, ...newDataRef };
-      } else {
-        return newDataRef;
-      }
-    });
-
-    /*
-     ** Delete dataRefs, add to cidPruneList
-     */
-    const deneutralizedPath = deneutralizePath(path, dataBucket?.payload?.cid);
-    const dataRefsToDelete = existingDataRefs.filter(
-      (e) => e.path.startsWith(deneutralizedPath + '/') || e.path === deneutralizedPath,
-    );
+    const entriesToDeleteIds = entriesToDelete.map((e) => e.id);
 
-    const dataRefDeletionIds = dataRefsToDelete.map((e) => e.id);
-    const formattedPruneList = dataRefsToDelete.map((e) => {
+    const formattedPruneList: Prisma.CidPruneListCreateManyInput[] = entriesToDelete.map((e) => {
       return {
-        description: '[DATA::DELETE]path: ' + neutralizePath(e.path),
+        description: '[DATA::DELETE db]path: ' + e.path,
         cid: e.cid,
-        type: e.type,
+        type: DataType.UNKNOWN,
         size: e.size,
         nodeId: e.nodeId,
         userId: e.userId,
         directory: e.directory,
       };
     });
 
-    const [deletions, creations, ...updates] = await prisma.$transaction([
-      prisma.dataReference.deleteMany({ where: { id: { in: dataRefDeletionIds } } }),
+    const existingDataRefs = await prisma.dataReference.findMany({
+      where: {
+        nodeId: node.id,
+        userId: owner.id,
+        type: { not: DataType.MANIFEST },
+      },
+    });
+
+    const dataRefsToDelete = existingDataRefs.filter((e) => e.path.startsWith(path + '/') || e.path === path);
+
+    const dataRefDeletionIds = dataRefsToDelete.map((e) => e.id);
```

Comment:
Personal preference: I think sequences of filter/map/reduce are much easier to follow as chains:

```suggestion
    const dataRefsToDelete = existingDataRefs
      .filter((e) => e.path.startsWith(path + '/') || e.path === path)
      .map((e) => e.id);
```